oauth2/oauth2-resource-server/src/main/java/org/springframework/security/oauth2/server/resource/authentication/OAuth2IntrospectionClaimNames.java

```diff
@@ -17,12 +17,13 @@
 
 /**
  * The names of the &quot;Introspection Claims&quot; defined by an
- * <a target="_blank" href="https://tools.ietf.org/html/rfc7662#section-2.2">Introspection Response</a>.
+ * <a target="_blank" href="https://tools.ietf.org/html/rfc7662#section-2.2">Introspection
+ * Response</a>.
  *
  * @author Josh Cummings
  * @since 5.2
  */
-interface OAuth2IntrospectionClaimNames {
+public interface OAuth2IntrospectionClaimNames {
```

Comment:
done

As a side note, it could be useful to create a ticket to expose it publicly and maybe in a lib higher in the stack, just like `JwtClaimNames` and `IdTokenClaimNames` are for instance

Reply:
Good call. This is something being considered, but we haven't run into a documented use case for introspection outside of the Resource Server. Do you have one?

Reply:
I do use it from introspection token builder as can be seen on the gh-6557--reviews branch

It could also be useful to solve https://github.com/spring-projects/spring-security/issues/6807 if you ever choose to access issue and expiration instants from attributes like I did

Reply:
So, the test support would also be in the context of a resource server. Or that is, we test our app for how we've configured it. You'd need a production use case for non-resource-server introspection to need test support at a different level of the stack.

My question is whether there are documented production use cases for an OAuth 2.0 client application calling an introspection endpoint (this is off-topic from this PR, of course, but just wanted to clarify).